### PR TITLE
Fix Inductor bench BC change

### PR DESCRIPTION
**Pull request by msaroufim**
Fixes https://github.com/pytorch/ao/issues/637

### CODE REVIEW

**Review by vkuzo — APPROVED** [review]

File: torchao/quantization/autoquant.py [hunk 1]
```diff
@@ -15,14 +15,19 @@
...
 try:
     from torch._inductor.utils import do_bench
-except:
-    from torch._inductor.runtime.runtime_utils import do_bench
+except ImportError:
+    try:
+        from torch._inductor.runtime.runtime_utils import do_bench
+    except ImportError:
+        from torch._inductor.runtime.benchmarking import benchmarker
+        do_bench = benchmarker.benchmark
```
Comment (nmacchioni) [inline torchao/quantization/autoquant.py:30]: Do we need these imports?

File: torchao/quantization/autoquant.py [hunk 2]
```diff
@@ -227,9 +232,13 @@ def do_autoquant_bench(op, *args, **kwargs):
         graph = torch.cuda.CUDAGraph()
         with torch.cuda.graph(graph, stream=stream):
             op(*args, **kwargs)
-        if TORCH_VERSION_AFTER_2_3:
+        if TORCH_VERSION_AFTER_2_3 and not TORCH_VERSION_AFTER_2_5:
             from torch._inductor.runtime.runtime_utils import do_bench_gpu
             res = do_bench_gpu(lambda: graph.replay(), warmup=warmup, rep=rep, return_mode="median")
+        elif TORCH_VERSION_AFTER_2_5 and torch.cuda.is_available():
```
Comment (nmacchioni) [inline torchao/quantization/autoquant.py:238]: Why `torch.cuda.is_available()`?

Comment (nmacchioni) [inline torchao/quantization/autoquant.py:238]: Should this be `TORCH_VERSION_AFTER_2_4`?

File: torchao/quantization/autoquant.py [hunk 3]
```diff
@@ -227,9 +232,13 @@ def do_autoquant_bench(op, *args, **kwargs):
...
+            from torch._inductor.runtime.benchmarking import benchmarker
+            do_bench_gpu = benchmarker.benchmark_gpu
+            res = do_bench_gpu(lambda: graph.replay(), warmup=warmup, rep=rep, return_mode="median")
```
Comment (nmacchioni) [inline torchao/quantization/autoquant.py:241]: The newest (i.e. `from torch._inductor.runtime.benchmarking import benchmarker`) block needs to go first, as from my understanding `TORCH_VERSION_AFTER_X_Y=True` always in fbcode.

**Comment by nmacchioni** [comment]
@msaroufim left some comments, I'm not sure if this resolves the BC changes (sorry if I'm just confused here on what is happening)

File: torchao/quantization/autoquant.py (hunk 1 above)
Comment (msaroufim) [reply torchao/quantization/autoquant.py:30]: AO needs to work over multiple pytorch versions and this api has been broken multiple times

We can explore getting rid of the usage completely @HDCharles

Comment (nmacchioni) [reply torchao/quantization/autoquant.py:30]: Can we not inline the imports in the IF/ELSE blocks later on when they get used (I see some already are?)?

**Pull request merged**